src/adapter/mod.rs

```diff
@@ -52,6 +52,13 @@ impl<'a> RustdocAdapter<'a> {
     pub fn schema() -> &'static Schema {
         &SCHEMA
     }
+
+    pub(crate) fn crate_at_origin(&self, origin: &Origin) -> Option<&'a PackageIndex<'a>> {
```

Comment:
Probably a good idea to suggest to the compiler that it inline this function.
```suggestion
    #[inline]
    pub(crate) fn crate_at_origin(&self, origin: &Origin) -> Option<&'a PackageIndex<'a>> {
```